### PR TITLE
Refactor to use Best Practices

### PR DESCRIPTION
**Pull request by WouterJ**
I think it would be good to use the Symfony Best Practices in example applications like this. I haven't changed any logic, just moving files, renaming, etc.

I haven't test anything yet, that's the main reason it's in WIP status at the moment.

### CODE REVIEW

File: src/TestBundle/TestBundle.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace TestBundle;
+
+use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+class TestBundle extends Bundle
```
Comment (WouterJ) [inline src/TestBundle/TestBundle.php]: Can anyone explain me why there is a TestBundle?

File: src/AppBundle/AppBundle.php [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?php
 
 /*
```
Comment (WouterJ) [inline src/AppBundle/AppBundle.php]: this is the only license header I could find in the project code. Should we add it to the other classes as well?

**Comment by WouterJ** [comment]
Hmm, so this PR can't be merged very soon because of https://github.com/sonata-project/SonataAdminBundle/pull/2594#issuecomment-74387408

File: src/AppBundle/AppBundle.php (hunk 2 above)
Comment (dbu) [inline src/AppBundle/AppBundle.php]: i guess we have been neglectant on these. yeah lets have them everywhere as usual.

File: src/TestBundle/TestBundle.php (hunk 1 above)
Comment (dbu) [inline src/TestBundle/TestBundle.php]: i think at some point, the idea was to have some demo things but looking at we have in this bundle vs the other one, i feel we could just move all to the AppBundle.

**Comment by dbu** [comment]
:+1: 

for sonata, see https://github.com/sonata-project/SonataAdminBundle/pull/2840

File: src/AppBundle/AppBundle.php (hunk 2 above)
Comment (lsmith77) [inline src/AppBundle/AppBundle.php]: +1

File: src/TestBundle/TestBundle.php (hunk 1 above)
Comment (lsmith77) [inline src/TestBundle/TestBundle.php]: +1

**Comment by lsmith77** [comment]
+1 to get this finished

**Comment by dbu** [comment]
@WouterJ i think if you update sonata admin in the composer.lock, this should now work. do you also want to eliminate the DemoBundle?

**Comment by WouterJ** [comment]
@dbu it doesn't work, as Sonata needs to release a bug fixing 2.3 release: https://github.com/sonata-project/SonataAdminBundle/pull/2840

And I don't think we should remove the AcmeDemoBundle, unless we make the sandbox very simple (it's now full-blown showing all possible features).

**Comment by dbu** [comment]
> And I don't think we should remove the AcmeDemoBundle, unless we make
> the sandbox very simple (it's now full-blown showing all possible features).

but right now, most of the demo code is in MainBundle, and just one
controller in DemoBundle. the sandbox is to show as many features as
possible, while the standard edition is to be kept minimal. so i don't
see why we would need two bundles.

**Comment by dbu** [comment]
i just tagged a release that has the fix: https://github.com/sonata-project/SonataAdminBundle/releases/tag/2.3.3

**Comment by WouterJ** [comment]
Thanks @dbu! I've now updated this PR, also fixing the discussed things. Imo, it's ready to merge if the tests pass.

**Comment by WouterJ** [comment]
Hmm, "symfony-cmf/sandbox 1.3.x-dev requires sonata-project/admin-bundle >=2.3.3 -> no matching package found". That's a weird error message, given 2.3.3 does exists.

**Comment by dbu** [comment]
this is bizarr. i just restarted, in case it would have been too close between releasing and packagist not updated, but its here. no idea what could be wrong - [packagist](https://packagist.org/packages/sonata-project/admin-bundle) sees a 2.3.3 but [travis](https://travis-ci.org/symfony-cmf/cmf-sandbox/jobs/58084495) not: symfony-cmf/sandbox 1.3.x-dev requires sonata-project/admin-bundle >=2.3.3 -> no matching package found. there are no further packages reported that would make sonata-project/admin-bundle uninstallable..

**Comment by WouterJ** [comment]
Same error... @Seldaek can you maybe give some hints on what's going wrong here?

**Comment by Seldaek** [comment]
```
$ c show sonata-project/admin-bundle
name     : sonata-project/admin-bundle
descrip. : Symfony SonataAdminBundle
keywords : admin, Admin Generator, sonata, bootstrap
versions : dev-master, 2.4.x-dev, 2.3.x-dev, 2.3.3, [...]
```

Looks good to me... version is there, no clue why it'd fail :/ Did you try locally to see if it's just travis? If yes then maybe they cache http requests too greedily or something? Are you using their new docker based build system?

**Comment by WouterJ** [comment]
Found out that there was a strange version constraint in SonataDoctrinePhpcrAdminBundle. Submitted a PR for it: https://github.com/sonata-project/SonataDoctrinePhpcrAdminBundle/pull/334

@Seldaek this seems like a problem that should get some nicer error messaging. With minimum-stability: stable, I got this error message:

    $ composer update
    Loading composer repositories with package information
    Updating dependencies (including require-dev)
    Your requirements could not be resolved to an installable set of packages.

    Problem 1
        - sonata-project/doctrine-phpcr-admin-bundle 1.2.1 requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> no matching package found.
        - sonata-project/doctrine-phpcr-admin-bundle 1.2.0 requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> no matching package found.
        - sonata-project/doctrine-phpcr-admin-bundle 1.2.1 requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> no matching package found.
        - Installation request for sonata-project/doctrine-phpcr-admin-bundle 1.2.* -> satisfiable by sonata-project/doctrine-phpcr-admin-bundle[1.2.0, 1.
    2.1].

    Potential causes:
    - A typo in the package name
    - The package is not available in a stable-enough version according to your minimum-stability setting
    see <https://groups.google.com/d/topic/composer-dev/_g3ASeIFlrc/discussion> for more details.

    Read <http://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.

It doesn't actually tell what's wrong, which is strange.

With stability: dev, I got this:

    Loading composer repositories with package information
    Updating dependencies (including require-dev)
    Your requirements could not be resolved to an installable set of packages.

    Problem 1
        - sonata-project/doctrine-phpcr-admin-bundle 1.2.0-RC1 requires sonata-project/admin-bundle >=2.2.9,<=2.2.12 -> no matching package found.
        - sonata-project/doctrine-phpcr-admin-bundle 1.2.1 requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> satisfiable by sonata-project/admin-bun
    dle[2.2.x-dev].
        - sonata-project/doctrine-phpcr-admin-bundle 1.2.0 requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> satisfiable by sonata-project/admin-bun
    dle[2.2.x-dev].
        - sonata-project/doctrine-phpcr-admin-bundle 1.2.0-RC2 requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> satisfiable by sonata-project/admin
    -bundle[2.2.x-dev].
        - sonata-project/doctrine-phpcr-admin-bundle 1.2.1 requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> satisfiable by sonata-project/admin-bun
    dle[2.2.x-dev].
        - sonata-project/doctrine-phpcr-admin-bundle 1.2.x-dev requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> satisfiable by sonata-project/admin
    -bundle[2.2.x-dev].
        - Conclusion: don't install sonata-project/admin-bundle 2.2.x-dev
        - Installation request for sonata-project/doctrine-phpcr-admin-bundle 1.2.* -> satisfiable by sonata-project/doctrine-phpcr-admin-bundle[1.2.0, 1.
    2.0-RC1, 1.2.0-RC2, 1.2.1, 1.2.x-dev].

    Potential causes:
    - A typo in the package name
    - The package is not available in a stable-enough version according to your minimum-stability setting
    see <https://groups.google.com/d/topic/composer-dev/_g3ASeIFlrc/discussion> for more details.

    Read <http://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.

Which somewhat tells what's wrong, but it's conclusion is wrong: "don't install sonata-project/admin-bundle 2.2.x-dev", as I'm trying to install 2.3.3...

**Comment by lsmith77** [comment]
So we have a new release of the SonataDoctrinePHPCRBundle and I retriggered travis, so why is it still failing?

**Comment by WouterJ** [comment]
@lsmith77 the problem here is not related to that. I have no idea what's going wrong here and I can't reproduce it locally...

**Comment by lsmith77** [comment]
Ah yeah .. now its complaining it cannot find SonataAdminBundle 2.3.3 .. yet https://packagist.org/packages/sonata-project/admin-bundle

**Comment by lsmith77** [comment]
you need to commit the updated composer.lock since we are using ``composer install --prefer-source`` to install the dependencies.

**Comment by WouterJ** [comment]
To see if it's a caching problem, I've deleted the composer cache from travis and retriggered the tests.

**Comment by WouterJ** [comment]
Hmm, this is when trying to run `composer update` locally:

    Problem 1
    - sonata-project/doctrine-phpcr-admin-bundle 1.2.2 requires sonata-project/admin-bundle ~2.4 -> no matching package found.
    - sonata-project/doctrine-phpcr-admin-bundle 1.2.1 requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> no matching package found.
    - sonata-project/doctrine-phpcr-admin-bundle 1.2.0 requires sonata-project/admin-bundle >=2.2.9,<=2.3.0 -> no matching package found.
    - Removal request for sonata-project/doctrine-phpcr-admin-bundle == 1.2.9999999.9999999-dev
    - Installation request for sonata-project/doctrine-phpcr-admin-bundle 1.2.* -> satisfiable by sonata-project/doctrine-phpcr-admin-bundle[1.2.0, 1.
    2.1, 1.2.2, 1.2.x-dev].

**Comment by WouterJ** [comment]
@lsmith77 seems like you tagged master as 1.2.2

**Comment by lsmith77** [comment]
argh. will fix asap

**Comment by lsmith77** [comment]
should be corrected now

**Comment by Seldaek** [comment]
@WouterJ I know the error reporting sometimes is off like this. I don't know why though and it's kinda hard to reproduce in tests since it seems to depend on wind speed and other weird factors :/

**Comment by lsmith77** [comment]
@WouterJ not it seems like the issue is in the test suite regarding the location of the fixture file

File: app/tests/HomepageTest.php [hunk 3]
```diff
@@ -23,7 +30,7 @@ public function testContents()
 
         $crawler = $client->request('GET', '/en');
 
-        $this->assertEquals(200, $client->getResponse()->getStatusCode());
+        $this->assertResponseSuccess($crawler->getResponse());
```
Comment (lsmith77) [inline app/tests/HomepageTest.php]: should be ``$client->getResponse()``

File: app/tests/StaticPageTest.php [hunk 4]
```diff
@@ -13,7 +20,7 @@ public function testContent($url, $title)
 
         $crawler = $client->request('GET', $url);
 
-        $this->assertEquals(200, $client->getResponse()->getStatusCode());
+        $this->assertResponseSuccess($crawler->getResponse());
```
Comment (lsmith77) [inline app/tests/StaticPageTest.php]: should be ``$client->getResponse()``

**Comment by WouterJ** [comment]
This PR is finally ready with passing tests!

**Pull request merged**